### PR TITLE
[netcore] Add System.Reflection.Emit support.

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

**Comment by vargaz** [comment]
With this, the SRE testsuite at least compiles and runs, with some failures:
RUN: 1938
FAILURES: 231
SKIPPED: 18

Note that only the bare minimum was done to make this work with netcore, a large amount of further cleanups are possible.

**Comment by EgorBo** [comment]
Tried to run BenchmarkDotNet:
```
User defined subclasses of System.Type are not yet supported.
```

When I try to run it using so-called "InProcess" toolchain it fails with:
```
Message: Specified cast is not valid.,
Stack trace:
   at System.Reflection.CustomAttributeExtensions.GetCustomAttributes(MemberInfo element)
```

**Comment by vargaz** [comment]
This is a work in progress, not everything is going to work.

File: mono/metadata/reflection.c [hunk 1]
```diff
@@ -1312,6 +1312,11 @@ method_body_object_construct (MonoDomain *domain, MonoClass *unused_class, MonoM
 	memcpy (il_data, header->code, header->code_size);
 	mono_gchandle_free_internal (il_gchandle);
 
+#ifdef ENABLE_NETCORE
+	g_assert_not_reached ();
```
Comment (EgorBo) [inline mono/metadata/reflection.c]: I am hitting this assert when I try to run xunit tests for System.Runtime.Tests using 
```
make run-xunit COREFX_ROOT=/prj/corefx-egor TEST_ASSEMBLY=/prj/corefx-egor/artifacts/bin/System.Runtime.Tests/netcoreapp-Unix-Debug/System.Runtime.Tests.dll
```

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by vargaz** [comment]
@monojenkins build failed

**Pull request merged**